packages/http-server/src/__tests__/body-params-validation.spec.ts

```diff
@@ -0,0 +1,277 @@
+import getHttpOperations from '@stoplight/prism-cli/src/util/getHttpOperations';
+import { createLogger } from '@stoplight/prism-core';
+import fastify = require('fastify');
+import { relative, resolve } from 'path';
+import { createServer } from '../';
+import { IPrismHttpServer } from '../types';
+
+const logger = createLogger('TEST', { enabled: false });
+
+async function instantiatePrism(fixture: string) {
+  const path = relative(process.cwd(), resolve(__dirname, 'fixtures', fixture));
+  const operations = await getHttpOperations(path);
+  const server = createServer(operations, { components: { logger }, config: { mock: { dynamic: false } } });
+  return server;
+}
+
+describe('body params validation', () => {
+  let server: IPrismHttpServer;
+
+  afterAll(async () => {
+    await server.fastify.close();
+  });
+
+  describe('oas3 with encodings', () => {
+    beforeEach(async () => {
+      server = await instantiatePrism(`encodings.oas3.yaml`);
+    });
+
+    // Ref: https://github.com/stoplightio/prism/issues/496
+    test.skip('allowReserved set to true', async () => {
+      const response = await server.fastify.inject({
+        method: 'POST',
+        url: '/allowReserved',
+        payload: {
+          reserved: ":/?#[]@!$&'()*+,;",
+        },
+      });
+
+      expect(response.statusCode).toEqual(422);
+    });
+  });
+
+  describe.each([['oas2'], ['oas3']])('%s with body param', oas => {
+    beforeEach(async () => {
+      server = await instantiatePrism(`operations-with-body-param.${oas}.yaml`);
+    });
+
+    describe('operation with no request content type defined', () => {
+      const operation: fastify.HTTPInjectOptions = {
+        method: 'POST',
+        url: '/json-body-no-request-content-type',
+      };
+
+      describe('property type invalid', () => {
+        test('returns 422 & error message', async () => {
+          const response = await server.fastify.inject({
+            ...operation,
+            payload: {
+              id: 'string',
+            },
+          });
+
+          expect(response.statusCode).toBe(422);
+          expect(JSON.parse(response.payload)).toMatchObject({
+            validation: [
+              {
+                code: 'type',
+                location: ['body', 'id'],
+                message: 'should be integer',
+                severity: 'Error',
+              },
+            ],
+          });
+        });
+      });
+    });
+
+    describe('operation with required property', () => {
+      const operation: fastify.HTTPInjectOptions = {
+        method: 'POST',
+        url: '/json-body-property-required',
```

Comment:
I'm afraid this test is wrong. The `required` property is false by default, and according to the definition file

```yaml
 "/json-body-property-required":
    post:
      consumes:
      - application/json
      parameters:
      - in: body
        name: body
        schema:
          type: object
          properties:
            id:
              type: integer
              format: int64
          required:
          - id
      responses:
        '200':
          schema:
            type: string
```

This should not return a validation error. Is this a problem in `http-spec`?

Reply:
@XVincentX not quite, the body itself is optional, but of present it must have the id property

Reply:
@XVincentX like @philsturgeon said. This test is checking if (when optional body is passed) we verify that a required **property** is passed.

I pass empty body `{}` but spec expects `{id:123}`. Therefore we get validation error - which is correct.